src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemService.ChangeState.cs

```diff
@@ -0,0 +1,115 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using FellowOakDicom;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Store;
+using Microsoft.Health.Dicom.Core.Features.Store.Entries;
+using Microsoft.Health.Dicom.Core.Features.Workitem.Model;
+using Microsoft.Health.Dicom.Core.Messages.Workitem;
+
+namespace Microsoft.Health.Dicom.Core.Features.Workitem;
+
+/// <summary>
+/// Provides functionality to process the list of <see cref="IDicomInstanceEntry"/>.
+/// </summary>
+public partial class WorkitemService
+{
+    public async Task<ChangeWorkitemStateResponse> ProcessChangeStateAsync(
+        DicomDataset dataset,
+        string workitemInstanceUid,
+        CancellationToken cancellationToken = default)
+    {
+        EnsureArg.IsNotNull(dataset, nameof(dataset));
+
+        try
+        {
+            var workitemMetadata = await _workitemOrchestrator
+                .GetWorkitemMetadataAsync(workitemInstanceUid, cancellationToken)
+                .ConfigureAwait(false);
+
+            if (workitemMetadata == null)
```

Comment:
Do we need this? Are we not throwing SQL exception when we dont find a workitem?